controller/server_controller.py

```diff
@@ -128,23 +130,35 @@ def delete_fn(labels, body, namespace, name, **_):
     """
     The juptyer server has been deleted.
     """
-    current_state = body.get("status", {}).get("state", ServerStatusEnum.Starting.value)
     api = get_api(config.api_version, config.custom_resource_name, config.api_group)
+    new_status = ServerStatusEnum.Stopping.value
+    if body:
+        old_status = body.get("status", {}).get("state")
+    else:
+        old_status = None
+    if old_status != new_status:
+        extra_labels = {
+            "status_from": str(old_status),
```

Comment:
Should we lower-case this? All the `ServerStatusEnum` values are lower-case. I also wonder if we want to have a dedicated `indeterminate` or `unknown` value instead of `none` that we could use here.


There's also the philosophical question on if going from no status to a status constitutes a status change (i.e. if we should count it at all), but then I think it doesn't make a difference. :slightly_smiling_face: